Reaktoro/Equilibrium/SmartEquilibriumSolver.cpp

```diff
@@ -247,13 +247,14 @@ struct SmartEquilibriumSolver::Impl
         for(auto i = 0; i < ips.size(); ++i)  // TODO: type Indices should be alias to VectorXi, to avoid such kind of codes
             ips[i] = ies[iorder[i]];
 
-        // The number of primary species among the equilibrium species
+        // The number of primary species among the equilibrium species (Np <= Ne)
         const auto& Np = canonicalizer.numBasicVariables();
 
         // Store the indices of primary and secondary species in state
         state.equilibrium().setIndicesEquilibriumSpecies(ips, Np);
 
         // The indices of the primary species at the calculated equilibrium state
+        // TODO: is this operation needed? Can't we access primary indices as const auto& iprimary = ips.head(Np);
```

Comment:
I think this is what this method call below does too. `ips.head(Np)` may be more preferable indeed.

Reply:
Corrected